src/object/srv_obj.c

```diff
@@ -363,6 +385,28 @@ obj_bulk_transfer(crt_rpc_t *rpc, crt_bulk_op_t bulk_op, bool bulk_bind,
 		rc = ret ? dss_abterr2der(ret) : *status;
 
 	ABT_eventual_free(&arg.eventual);
+	/* After RDMA is done, corrupt the server data */
+	if (DAOS_FAIL_CHECK(DAOS_CHECKSUM_SDATA_CORRUPT)) {
+		struct obj_rw_in	*orw = crt_req_get(rpc);
+		struct ds_pool		*pool;
+
+		pool = ds_pool_lookup(orw->orw_pool_uuid);
+		if (pool == NULL)
+			return -DER_NONEXIST;
```

Comment:
Maybe just D_ASSERT(pool != NULL), since pool must exist here during I/O

Reply:
Can I update this later while I work other issues?